frontend/src/app/app.tsx

```diff
@@ -1,56 +1,35 @@
-import reactLogo from '~/assets/img/react.svg';
 import { actions as authActions } from '~/bundles/auth/store/auth.js';
 import {
-    Link,
     Loader,
     RouterOutlet,
 } from '~/bundles/common/components/components.js';
-import { AppRoute, DataStatus } from '~/bundles/common/enums/enums.js';
 import {
     useAppDispatch,
-    useAppSelector,
     useEffect,
-    useLocation,
+    useState,
 } from '~/bundles/common/hooks/hooks.js';
 
 const App: React.FC = () => {
-    const { pathname } = useLocation();
+    const [isRefreshing, setIsRefreshing] = useState(true);
     const dispatch = useAppDispatch();
-    const { dataStatus } = useAppSelector(({ auth }) => ({
-        isRefreshing: auth.isRefreshing,
-        dataStatus: auth.dataStatus,
-    }));
 
     useEffect(() => {
-        void dispatch(authActions.refreshUser());
+        const refreshUser = async (): Promise<void> => {
+            try {
+                await dispatch(authActions.refreshUser());
+            } finally {
+                setIsRefreshing(false);
+            }
+        };
+
+        void refreshUser();
     }, [dispatch]);
 
-    if (dataStatus === DataStatus.PENDING) {
+    if (isRefreshing) {
```

Comment:
why do we need to change it?

Reply:
https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/112903531/81491a16-d6cf-4d44-9378-ed0de4ff167a

Because it is used in register page for loader in button and if I add it, there would be something like that